hdat/hdat_cli.py

```diff
@@ -113,3 +130,55 @@ def diff_results(suites, golden_result, result):
         traceback.print_exc()
         msg = 'Error when attempting to show "{}": {}'
         raise AbortError(msg.format(print_resultspec(result), e))
+
+
+def get_result_data(result, input_key_list):
+    keys = []
+    data = []
+    unknown_keys = []
+    found_nested_value = False
+
+    for in_key in input_key_list:
+        if ".*" in in_key:
+            nested_in_key = in_key.replace(".*", "")
+            for result_key in result.keys():
```

Comment:
I think the sorting issues on the CI for Python 3.4 and 3.5 are coming from here (and similar in line 144) as `dict` keys aren't in a predictable order. I think putting `sorted(result.keys())` here and also calling `input_key_list.sort()` at the top of this method should fix it. 

Also, running `tox` before pushing can be used to debug issues on multiple Python versions, like this or using language features not present in older versions.